packages/Discretization/src/DTK_PointSearch_def.hpp

```diff
@@ -0,0 +1,764 @@
+        Kokkos::Experimental::MinMaxScalar<unsigned int> result;
+        Kokkos::Experimental::MinMax<unsigned int> reducer( result );
+        parallel_reduce(
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_imports ),
+            Kokkos::Impl::min_max_functor<
+                Kokkos::View<unsigned int *, DeviceType>>( imported_query_ids ),
+            reducer );
+        if ( result.min_val != result.max_val )
+        {
+            Kokkos::BinSort<Kokkos::View<unsigned int *, DeviceType>, CompType>
+                bin_sort(
+                    imported_query_ids,
+                    CompType( n_imports / 2, result.min_val, result.max_val ),
+                    true );
+            bin_sort.create_permute_vector();
+            bin_sort.sort( imported_query_ids );
+            bin_sort.sort( imported_cell_indices );
+            bin_sort.sort( imported_ranks );
+            bin_sort.sort( imported_ref_pts );
+        }
+
+        // Compute the offset. We could to this on the device but it is more
+        // complex and we will always need to do a communication from the GPU to
+        // the CPU to allocate memory for the offset_view.
+        auto imported_query_ids_host =
+            Kokkos::create_mirror_view( imported_query_ids );
+        std::vector<unsigned int> offset;
+        offset.push_back( 0 );
+        for ( unsigned int i = 1; i < imported_query_ids.extent( 0 ); ++i )
+        {
+            if ( imported_query_ids_host( i ) !=
+                 imported_query_ids_host( i - 1 ) )
+                offset.push_back( i );
+        }
+        offset.push_back( imported_query_ids.extent( 0 ) );
+        Kokkos::realloc( offset_view, offset.size() );
+        auto offset_view_host = Kokkos::create_mirror_view( offset_view );
+        for ( unsigned int i = 0; i < offset.size(); ++i )
+            offset_view_host( i ) = offset[i];
+        Kokkos::deep_copy( offset_view, offset_view_host );
+    }
+    else
+    {
+        Kokkos::realloc( offset_view, 1 );
+        Kokkos::deep_copy( offset_view, 0 );
+    }
+
+    return std::make_tuple( imported_ranks, imported_cell_indices,
+                            imported_ref_pts, offset_view );
+}
+
+template <typename DeviceType>
+void PointSearch<DeviceType>::convertMesh(
+    std::array<unsigned int, DTK_N_TOPO> const &n_cells_per_topo,
+    Kokkos::View<DTK_CellTopology *, DeviceType> const &cell_topologies_view,
+    Kokkos::View<unsigned int *, DeviceType> const &cells,
+    Kokkos::View<double **, DeviceType> const &coordinates )
+{
+    // First, we get the number of nodes for each topology to get initialize
+    // _block_cells at the right size.
+    Kokkos::View<unsigned int[DTK_N_TOPO], DeviceType> n_nodes_per_topo(
+        "n_nodes_per_topo" );
+    auto n_nodes_per_topo_host = Kokkos::create_mirror_view( n_nodes_per_topo );
+    Topologies topologies;
+    for ( int i = 0; i < DTK_N_TOPO; ++i )
+        n_nodes_per_topo_host( i ) = topologies[i].n_nodes;
```

Comment:
You guys are welcomed to change `Topologies` but `Topologies` is an object because it contains data. There are work around of course but I really don't like them because I have had similar code that stop working when I used `lto`. Always great to wait a few hours for the compiler to optimize your code and then have it crash on your test suite...

Reply:
We should talk about this one

Reply:
> `Topologies` is an object because it contains data

data could be declared static

Reply:
Like I said just before I know and I have already hit by bug including in Cap when using static data and lto

Reply:
> LTO

link time optimization?

Reply:
yeah